url.go

```diff
@@ -152,64 +157,106 @@ func ParseReference(url string) (*Reference, error) {
 
 func parseReference(url string) (*Reference, error) {
 	var r Reference
-	i := strings.Index(url, ":")
-	if i >= 0 {
-		r.Schema = url[:i]
-		if r.Schema != "cs" && r.Schema != "local" {
-			return nil, fmt.Errorf("entity URL has invalid schema: %q", url)
+	// Check if we're dealing with a v1 or v2 URL.
+	// TODO: this is a naive check that only works on fully-qualified URLs;
+	// further work will be required for future URLs
+	// XXX Get rid of urlVersion, test based on validStore.MatchString
+	if validStore.MatchString(url) {
+		r.Schema = "cs"
+		url = validStore.ReplaceAllString(url, "")
+		parts := strings.Split(url, "/")
```

Comment:
What about something like `parts := strings.Split(strings.Trim(url), "/")` so that we don't need to strip leading/trailing slashes later?